backend/cmd/headlamp.go

```diff
@@ -307,6 +314,35 @@ func createHeadlampHandler(config *HeadlampConfig) http.Handler {
 		proxy.ServeHTTP(w, r)
 	})
 
+	//nolint:gosec,scopelint
+	r.HandleFunc("/config/clusters/refresh", func(w http.ResponseWriter, r *http.Request) {
+		contexts := getContextFromKubeConfig(config.kubeConfigPath)
+
+		refreshedClusterProxies := 0
+
+		for _, context := range contexts {
```

Comment:
I don't think this handles contexts being removed?

What about if the context changes, but the name remains the same as an old one?

Reply:
Removing contexts can be quite tricky because now we support dynamic cluster. I will make changes to replace the ones with same name.

Reply:
It's not clear to me what you changed?

If it's not handling any of the cases, and can't be fixed... then can you please document what it can and can't do?

Reply:
Added comments. Earlier there wasn't a way to differentiate if a context was created from kubeConfig or Dynamic Cluster request or For InCluster. Now I have added a "source" key to the context struct. And all contexts with the "KUBECONFIG"  source are dropped and the new contexts are set up for the clusters found in the kubeConfig.